src/Momento.Sdk/Config/TopicConfiguration.cs

```diff
@@ -13,15 +14,20 @@ public class TopicConfiguration : ITopicConfiguration
     /// <inheritdoc />
     public ITopicTransportStrategy TransportStrategy { get; }
 
+    /// <inheritdoc />
+    public ISubscriptionRetryStrategy SubscriptionRetryStrategy { get; }
+
     /// <summary>
     /// Create a new instance of a Topic Configuration object with provided arguments: <see cref="Momento.Sdk.Config.ITopicConfiguration.TransportStrategy"/>, and <see cref="Momento.Sdk.Config.ITopicConfiguration.LoggerFactory"/>
     /// </summary>
     /// <param name="transportStrategy">This is responsible for configuring network tunables.</param>
     /// <param name="loggerFactory">This is responsible for configuring logging.</param>
-    public TopicConfiguration(ILoggerFactory loggerFactory, ITopicTransportStrategy transportStrategy)
+    /// <param name="subscriptionRetryStrategy">Configures whether and when to resubscribe to a topic after an error occurs.</param>
+    public TopicConfiguration(ILoggerFactory loggerFactory, ITopicTransportStrategy transportStrategy, ISubscriptionRetryStrategy? subscriptionRetryStrategy = null)
     {
         LoggerFactory = loggerFactory;
         TransportStrategy = transportStrategy;
+        SubscriptionRetryStrategy = subscriptionRetryStrategy ?? new DefaultSubscriptionRetryStrategy(loggerFactory.CreateLogger<DefaultSubscriptionRetryStrategy>());
```

Comment:
I would just add a constructor alternative to `DefaultSubscriptionRetryStrategy` that takes a loggerFactory as input and creates the logger internally.